pytorch_lightning/accelerators/tpu_accelerator.py

```diff
@@ -217,6 +205,15 @@ def __setup_tpu_training(self, model: LightningModule, trainer):
         trainer.global_rank = trainer.tpu_local_core_rank
         rank_zero_only.rank = trainer.global_rank
 
+        # if given an ordinal device, use this as the device
+        if trainer.tpu_id is not None:
+            tpu_device = xm.xla_device(trainer.tpu_id)
+        else:
+            tpu_device = xm.xla_device()
+        # track the device and move model to it
+        trainer._device = tpu_device
+        model.to(trainer._device)
+
```

Comment:
what is the reason, seems that the code above doe does not do any action, just getters...

Reply:
I have moved the code which sets the local and global rank above the code for moving the model to the device. Elsewhen we call `rank_zero_warn`, the rank is not set and all the processes print the warning.